  controller_manager:
    ros__parameters:
      update_rate: 30
      use_sim_time: false

      mec_dr_cont:
        type: "diff_drive_controller/DiffDriveController"
        ros__parameters:
          left_wheel_names: ["wheel_left_joint"]
          right_wheel_names: ["wheel_right_joint"]

          wheel_separation: 0.34           # Distance between wheels (meters)
          wheel_radius: 0.05               # Radius of the wheels (meters)
          
          cmd_vel_timeout: 0.5             # Timeout for velocity commands (seconds)

          use_stamped_vel: false           # Use stamped velocity commands (optional)

          enable_odom_tf: true             # Publish odom->base_link TF
          publish_rate: 50.0               # Rate (Hz) to publish odometry messages and TF

          pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.0, 0.0, 0.01]
          twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.0, 0.0, 0.01]

          # odom_frame_id and base_frame_id should match your TF frames
          odom_frame_id: "odom"
          base_frame_id: "base_link"

      joint_state_broadcaster:
        type: joint_state_broadcaster/JointStateBroadcaster



  # joint_trajectory_controller:
  #   ros__parameters:
  #     joints:
  #       - wheel_joint_fl
  #       - wheel_joint_fr
  #       - wheel_joint_fl
  #       - wheel_joint_fl


  #     command_interfaces:
  #       - position

  #     state_interfaces:
  #       - position
  #       - velocity

  #     action_monitor_rate: 20.0

  #     allow_partial_joints_goal: false
  #     interpolate_from_desired_state: true
  #     constraints:
  #       stopped_velocity_tolerance: 0.01
  #       goal_time: 0.0
  #       joint1:
  #         trajectory: 0.05
  #         goal: 0.03



    mec_dr_cont:
      ros__parameters:
        reference_timeout: 0.5 #0.1
      


        front_left_wheel_command_joint_name: "wheel_joint_fl"
        front_right_wheel_command_joint_name: "wheel_joint_fr"
        rear_right_wheel_command_joint_name: "wheel_joint_br"
        rear_left_wheel_command_joint_name: "wheel_joint_bl"
        
        

        front_left_wheel_state_joint_name: "wheel_joint_fl"
        front_right_wheel_state_joint_name: "wheel_joint_fr"
        rear_right_wheel_state_joint_name: "wheel_joint_br"
        rear_left_wheel_state_joint_name: "wheel_joint_bl"

        kinematics:
          base_frame_offset: { x: 0.0, y: 0.0, theta: 0.0 }
          wheels_radius: 0.05
          sum_of_robot_center_projection_on_X_Y_axis: 0.3855  
          #sum_of_robot_center_projection_on_X_Y_axis: 0.386  


        publish_odom: true
        enable_odom_tf: true
        publish_odom_tf: true
        odom_frame_id: odom
        base_frame_id: base_link
      
        twist_covariance_diagonal: [0.0, 7.0, 14.0, 21.0, 28.0, 35.0]
        pose_covariance_diagonal: [0.0, 7.0, 14.0, 21.0, 28.0, 35.0]


